pkg/virt-handler/vm.go

```diff
@@ -1271,10 +1271,21 @@ func (d *VirtualMachineController) checkNetworkInterfacesForMigration(vmi *v1.Vi
 	for _, network := range vmi.Spec.Networks {
 		networks[network.Name] = network.DeepCopy()
 	}
+	// if no networks or interfaces are found, return error
+	// this can be in the case of a new minimal VMI
+	if len(networks) == 0 || len(vmi.Spec.Domain.Devices.Interfaces) == 0 {
+		return fmt.Errorf("cannot migrate VMI, network or interfaces not found")
+	}
 	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if iface.Bridge != nil && networks[iface.Name].Pod != nil {
```

Comment:
@vatsalparekh how about just this one code change at all:

```go
if iface.Masquerade == nil && networks[iface.Name].Pod != nil {
    return fmt.Errorf("cannot migrate VMI which does not use masquerade to connect to the pod network")
}
```

Reply:
changed to this now